controlplane/kubeadm/internal/control_plane.go

```diff
@@ -124,6 +124,16 @@ func (c *ControlPlane) MachineInFailureDomainWithMostMachines(machines Filterabl
 	return machineToMark, nil
 }
 
+// MachineWithDeleteAnnotation returns a machine that has been annotated with DeleteMachineAnnotation key.
+func (c *ControlPlane) MachineWithDeleteAnnotation(machines FilterableMachineCollection) (*clusterv1.Machine) {
+	// See if there are any machines with DeleteMachineAnnotation key.
+	annotatedmachines := machines.Filter(machinefilters.HasAnnotationKey(clusterv1.DeleteMachineAnnotation))
+	// If there are, then pick the oldest machine from the filtered machines.
+	machineToPick := annotatedmachines.Oldest()
```

Comment:
Hm, could we instead return the whole list of annotated machines here? And let the logic above pick the one in the failure domain with most machines?

Reply:
@vincepri we were discussing this in the issue itself and kind of agreed to return the oldest machine first. Are there any objections/drawbacks of don't it this way? would you mind elaborating more if so?  Thanks

Reply:
`MachineInFailureDomainWithMostMachines` (what a mouthful 😂), picks the Machine that's in the failure domain (e.g. availability zone on AWS) with the highest count of machines in it, and then picks the oldest one from that list